api/handlers/ntlm.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/// <reference path="../../node/node.d.ts"/>
+
+import VsoBaseInterfaces = require('../interfaces/common/VsoBaseInterfaces');
+
+import http = require("http");
+import https = require("https");
+var _ = require("underscore");
+var ntlm = require("../opensource/node-http-ntlm/ntlm");
+
+export class NtlmCredentialHandler implements VsoBaseInterfaces.IRequestHandler {
+    username: string;
+    password: string;
+    workstation: string;
+    domain: string;
+
+    constructor(username: string, password: string,  domain?: string, workstation?: string) {
+        this.username = username;
+        this.password = password;
+        if (domain !== undefined) {
+            this.domain = domain;
+        }
+        if (workstation !== undefined) {
+            this.workstation = workstation;
+        }
+    }
+
+    prepareRequest(options:any): void {
+        // No headers or options need to be set.  We keep the credentials on the handler itself.
+    }
+
+    canHandleAuthentication(headers: any): boolean {
+        // Ensure that we're talking NTLM here
+        // Once we have the www-authenticate header, split it so we can ensure we can talk NTLM
+        var wwwAuthenticate = headers['www-authenticate'];
+        if (wwwAuthenticate !== undefined) {
+            var mechanisms = wwwAuthenticate.split(', ');
+            var idx =  mechanisms.indexOf("NTLM");
+            if (idx >= 0) {
+                // Check specifically for 'NTLM' since www-authenticate header can also contain
+                // the Authorization value to use in the form of 'NTLM TlRMTVNT....AAAADw=='
+                if (mechanisms[idx].length == 4) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    // The following method is an adaptation of code found at https://github.com/SamDecrock/node-http-ntlm/blob/master/httpntlm.js
+    handleAuthentication(httpClient, protocol, options, objs, finalCallback): void {
+        // Set up the headers for NTLM authentication
+        var ntlmOptions = _.extend(options, {
+            username: this.username,
+            password: this.password,
+            domain: (this.domain !== undefined) ? this.domain : '',
```

Comment:
You can do 
domain: this.domain || ''
If this.domain is null or undefined (or an empty string, false, or zero) js will use the second half of the condition.